log4j-api/src/main/java/org/apache/logging/log4j/spi/AbstractLogger.java

```diff
@@ -2755,77 +2755,114 @@ public void warn(final String message, final Object p0, final Object p1, final O
      * @since 3.0
      */
     @Override
-    public  LogBuilder atTrace() {
+    public  LogBuilder trace() {
         if (isTraceEnabled()) {
-            return logBuilder.get().setLevel(Level.TRACE);
+            DefaultLogBuilder builder = logBuilder.get();
+            if (builder.isInUse()) {
+                return new DefaultLogBuilder(this);
+            }
+            return builder.setLevel(Level.TRACE);
         } else {
-            return LogBuilder.INSTANCE;
+            return LogBuilder.NOOP;
         }
     }
     /**
-     * Constuct a trace log event.
+     * Constuct a debug log event.
      * @return a LogBuilder.
      * @since 3.0
      */
     @Override
-    public LogBuilder atDebug() {
+    public LogBuilder debug() {
         if (isDebugEnabled()) {
+            DefaultLogBuilder builder = logBuilder.get();
+            if (builder.isInUse()) {
+                return new DefaultLogBuilder(this);
+            }
             return logBuilder.get().setLevel(Level.DEBUG);
         } else {
-            return LogBuilder.INSTANCE;
+            return LogBuilder.NOOP;
```

Comment:
The above 7 lines are repeated for each log level. Would it be possible to extract this to a helper method?

Reply:
Isn’t that just replacing one line with another?

Reply:
There is no need for a new method. I have modified them to all call atLevel().